torchtext/experimental/models/roberta_model.py

```diff
@@ -0,0 +1,74 @@
+import os
+import torch
+import torch.nn as nn
+from torchtext.experimental.modules import TransformerEncoder
+from .roberta_transform import RobertaTransform
+from torchtext.utils import download_from_url, extract_archive, files_exist, load_args_from_json
+
+
+class RobertaModel(nn.Module):
+    """Contain a Roberta encoder model."""
+
+    def __init__(self, args, encoder):
+        super(RobertaModel, self).__init__()
+        self.args = args
+        self.encoder = encoder
+        self.transform = None
+
+    @classmethod
+    def build_model(self, args):
+        encoder = TransformerEncoder(args.ntoken, args.embed_dim, args.nhead,
+                                     args.feedforward_dim, args.nlayers, args.dropout)
+        return cls(args, encoder)
+
+    def load_transform(self, directory='./', tokenizer_file="sentencepiece.bpe.model", vocab_file='vocab.txt'):
+        self.transform = RobertaTransform.from_pretrained(directory=directory,
+                                                          tokenizer_file=tokenizer_file,
+                                                          vocab_file=vocab_file)
+
+    # [TODO] use torch.hub.load_state_dict_from_url
+    @classmethod
+    def from_pretrained(cls, args, directory='./', checkpoint_file="model.pt"):
+        encoder = TransformerEncoder(args.ntoken, args.embed_dim, args.nhead,
```

Comment:
It's not clear to me what is the benefit of having a `from_pretrained` method.
It's hard to enforce that the `args` correspond to the `checkpoint_file`, and it can give the user the expectation that it can magically work, but the user will just hit opaque error messages (like "sizes don't match for mha1.weight").

If we were to make the `from_pretrained` more robust, one would need to save the `args` inside the `checkpoint_file`, and use it instead to build the model